dnf5-plugins/copr_plugin/json.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "json.hpp"
+
+#include "tempfile.hpp"
+
+#include "libdnf/repo/file_downloader.hpp"
+
+#include <fstream>
+
+Json::Json(libdnf::Base & base, const std::string & url) {
+    auto temp_file = TempFile();
+    libdnf::repo::FileDownloader downloader(base);
```

Comment:
Do we have a better "download" support in DNF5?  I dislike the way the progress bar touches stdout.